pkg/providers/apisix/apisix_consumer.go

```diff
@@ -269,6 +272,9 @@ func (c *apisixConsumerController) onUpdate(oldObj, newObj interface{}) {
 	if !c.namespaceProvider.IsWatchingNamespace(key) {
 		return
 	}
+	if !c.isEffective(curr) {
```

Comment:
When I communicated with @tao12345666333  on wechat, in the `onUpdate` function, the value of `IngressClassName` might be changed from apisix to xxx, which should generate a delete event.
There are similar issues with other crd implementations such as apisixupstream. Maybe we can create a new PR focus to solve this problem.

Reply:
Yes. I have created an issue for tracking https://github.com/apache/apisix-ingress-controller/issues/1719 
We can address this in a follow-up PR